From 4f02440ffd7c8653085e685db0a356529faf2e16 Mon Sep 17 00:00:00 2001
From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Tue, 8 Mar 2011 18:13:34 +0100
Subject: ASoC: ABE HAL: Fixup initialization order inside abe_init_io_task

Due to potential race condition between ABE and MPU the
initialization order for the task descriptor and task process need
to be change to avoid potential memory corruption.

Change-Id: Ia40914025b1ca9bd85138ca47d11ce931ce38e5f
Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
---
 sound/soc/omap/abe/abe_ini.c |   44 ++++++++++++++++++++---------------------
 1 files changed, 21 insertions(+), 23 deletions(-)

Index: kernel/sound/soc/omap/abe/abe_ini.c
===================================================================
--- kernel.orig/sound/soc/omap/abe/abe_ini.c	2011-04-05 21:57:23.000000000 +0900
+++ kernel/sound/soc/omap/abe/abe_ini.c	2011-04-06 00:15:58.000000000 +0900
@@ -447,7 +447,7 @@
 		desc_third_word[2] = irq_dmareq_field;
 		abe_block_copy(COPY_FROM_HOST_TO_ABE, ABE_DMEM,
 			       sio_desc_address + struct_offset,
-			       (u32 *) desc_third_word, 4);
+			       (u32 *) desc_third_word, 4);   
 	} else {
 		/* serial interface: sync ATC with Firmware activity */
 		sio_desc_address =
